### PR TITLE
Ps 8.0 5754

### PR DESCRIPTION
**Pull request by patrickbirch**
Merged with 5.7 version and fixed the conflict

### CODE REVIEW

File: doc/source/management/data_at_rest_encryption.rst [hunk 1]
```diff
@@ -585,7 +583,46 @@ on every other key rotation.
...
 full text search using parallel sorting, building indexes using merge sort, and
-online DDL logs created by InnoDB for online DDL. Encryption is available for file merges used in queries and backend processes.
+online DDL logs created by InnoDB for online DDL.
+
+.. _data-at-rest-encryption.undo-tablespace:
+
+InnoDB Undo Tablespace Encryption
+================================================================================
+
+:Availability: This feature is **Experimental** quality
```
Comment (dutow) [inline doc/source/management/data_at_rest_encryption.rst:595]: This is an upstream feature for 8.0, and GA.

It is also possible to remove undo encryption in 8.0, by truncating the undo tablespaces.

File: doc/source/management/data_at_rest_encryption.rst [hunk 2]
```diff
@@ -585,7 +585,42 @@ on every other key rotation.
...
 full text search using parallel sorting, building indexes using merge sort, and
-online DDL logs created by InnoDB for online DDL. Encryption is available for file merges used in queries and backend processes.
+online DDL logs created by InnoDB for online DDL.
+
+.. _data-at-rest-encryption.undo-tablespace:
+
+InnoDB Undo Tablespace Encryption
+================================================================================
+
+:Availability: This feature is **Alpha** quality
```
Comment (dutow) [inline doc/source/management/data_at_rest_encryption.rst]: I still see alpha here, when according to our encryption spreadsheet, it is GA in 8.0.16?

File: doc/source/management/data_at_rest_encryption.rst [hunk 3]
```diff
@@ -585,7 +585,42 @@ on every other key rotation.
...
+
+The encryption of InnoDB Undo tablespaces is only available when using
```
Comment (dutow) [inline doc/source/management/data_at_rest_encryption.rst:597]: Sorry, I didn't notice this before: in 8.0, undo can no longer be part of the system tablespace, there is always at least 2 separate undo tablespace. See the upstream undo docs.

File: doc/source/management/data_at_rest_encryption.rst [hunk 4]
```diff
@@ -585,7 +585,42 @@ on every other key rotation.
...
+   :scope: Global
+   :vartype: Boolean
+   :default: ``Off``
+
+Enables the encryption of InnoDB Undo tablespaces. You can enable encryption and
+disable encryption while the server is running.
+
+.. note::
+
+    You can ``truncate the Undo tablespace <https://dev.mysql.com/doc/refman/8.0/en/innodb-undo-tablespaces.html#Truncating-Undo-Tablespaces>``__ to remove the encryption. When you enable undo log encryption, the server writes encryption information into the header and the information stays in the header during the life of the undo log. If you restart the server, the server will try to load the encryption key from the keyring during startup. If the keyring is not available, the server cannot start.
```
Comment (dutow) [inline doc/source/management/data_at_rest_encryption.rst]: The first sentence now says that you can remove the encryption, and the second that it stays there forever.

**Review by dutow — CHANGES_REQUESTED** [review]

File: doc/source/management/data_at_rest_encryption.rst (hunk 4 above)
Comment (patrickbirch) [reply doc/source/management/data_at_rest_encryption.rst]: Good point. I will emphasize the "disable" doesn't work.

File: doc/source/management/data_at_rest_encryption.rst (hunk 2 above)
Comment (patrickbirch) [reply doc/source/management/data_at_rest_encryption.rst]: Odd. For some reason, it didn't pull in the latest version. Fixed.

**Comment by patrickbirch** [comment]
Close this pull request - updated in the encryption doc update.

**Pull request closed without merging**